index.js

```diff
@@ -1,17 +1,29 @@
 'use strict'
 
 const fp = require('fastify-plugin')
+const process = require('process')
 const parallel = require('fastparallel')()
 
+const registeredListeners = []
+
 function fastifyGracefulShutdown(fastify, opts, next) {
   const logger = fastify.log.child({ plugin: 'fastify-graceful-shutdown' })
   const handlers = []
   const timeout = opts.timeout || 10000
   const signals = ['SIGINT', 'SIGTERM']
+  const handlerEventListener = opts.handlerEventListener || process
+
+  // Remove preexisting listeners if already created by previous instance of same plugin
+  if (opts.resetHandlersOnInit) {
+    registeredListeners.forEach(({ signal, listener }) => {
+      handlerEventListener.removeListener(signal, listener)
+    })
+    registeredListeners.splice(0, registeredListeners.length)
```

Comment:
Why not just `= []` ?

Reply:
I generally prefer immutable variables, but it's a tiny thing. Changed.